src/core/obj.js

```diff
@@ -1147,8 +1148,33 @@ var XRef = (function XRefClosure() {
               uncompressed: true,
             };
           }
-          var contentLength = skipUntil(buffer, position, endobjBytes) + 7;
-          var content = buffer.subarray(position, position + contentLength);
+          let contentLength, startPos = position + token.length;
+
+          // Find the next "obj" string, rather than "endobj", to ensure that
+          // we won't skip over a new 'obj' operator in corrupt files where
+          // 'endobj' operators are missing (fixes issue9105_reduced.pdf).
+          while (startPos < buffer.length) {
+            let endPos = startPos + skipUntil(buffer, startPos, objBytes) + 4;
+            contentLength = endPos - position;
+
+            let tokenStr = bytesToString(buffer.subarray(endPos - 8, endPos));
+            if (endobjRegExp.test(tokenStr)) {
+              break; // Found 'endobj'.
+            } else {
+              let objStartPos = Math.max(endPos - 25, startPos);
+              let objStr = bytesToString(buffer.subarray(objStartPos, endPos));
+
+              let objToken = /\d+\s+\d+\s+obj[\b\s]$/.exec(objStr);
```

Comment:
There is an `objRegExp` above which looks very similar. Shouldn't we use that? If not, can we cache this regex?

Reply:
The operative work is "similar", not identical ;-)
So no, we cannot use that one since we need to check when *end* of a string here (as opposed to the beginning above), and I'm not sure if caching really matters since this particular code-patch should almost never be hit in practice.